eng/pipelines/templates/jobs/archetype-sdk-client.yml

```
@@ -18,6 +18,10 @@ parameters:
       OSVmImage: "ubuntu-18.04"
       NodeTestVersion: "12.x"
```

For the browser tests, instead of hardcoding a node version I think it would make more sense to use the "default" version, since we are really testing the browser instead of node.  Do you know if matrixes can reference other variables?  For example, would this work?

```suggestion
      NodeTestVersion: "$(NodeVersion)"
```

---

An example of this exists in the [smoke test pipeline](https://github.com/Azure/azure-sdk-for-js/blob/master/common/smoke-test/smoke-tests.yml#L49-L54)

```
      - template: ../../eng/pipelines/templates/steps/common.yml

      - task: NodeTool@0
        displayName: Use Node $(NodeVersion)
        inputs:
          versionSpec: $(NodeVersion)
```

---

@benbp: Thanks for the pointer, but the question is whether this works in a `Matrix` definition.  Some parts of DevOps YAML files are processed in different ways and support different features (it's complicated).

---

> Do you know if matrixes can reference other variables? For example, would this work?

Yes I do believe they can reference other variables if they are defined higher in the variable scope (i.e. either at the global, stage, or job scope. Although I couldn't find any definitive answer when reading the docs at https://docs.microsoft.com/en-us/azure/devops/pipelines/process/variables?view=azure-devops&tabs=yaml%2Cbatch#macro-syntax-variables, so I guess we will have to test.